source/multigrid/mg_transfer_matrix_free.cc

```diff
@@ -355,6 +355,30 @@ MGTransferMatrixFree<dim, Number>::do_prolongate_add(
     Utilities::fixed_power<dim>(n_child_dofs_1d);
   constexpr unsigned int three_to_dim = Utilities::pow(3, dim);
 
+  // If we have user defined MG constraints, we must create
+  // a non-const, ghosted version of the source vector to distirbute
+  // constraints.
+  const LinearAlgebra::distributed::Vector<Number> *to_use = &src;
+  LinearAlgebra::distributed::Vector<Number>        new_src;
+  if (this->mg_constrained_dofs != nullptr &&
```

Comment:
```suggestion
  if ((*this->mg_constrained_dofs) != nullptr &&
```

Reply:
We get the following compile error with this change:

![error1](https://user-images.githubusercontent.com/11428698/64924903-88770800-d7b7-11e9-82ec-b0374be7eb30.png)

Reply:
@tjhei why do you want to dereference the pointer?